### PR TITLE
Use large composed DFG instead of smaller dfgs for each basic block

### PR DESCRIPTION
**Pull request by fazerlicourice7**
Update sim to use composed dfg.
Update instrumentation and compilation to keep track of var indices.
Merge in Cacti and integrate Mem and Buf vars with inverse pass. (doesn't yet inverse pass through cacti)
Add in bottleneck based architecture search.

### CODE REVIEW

File: src/schedule.py
Comment (fazerlicourice7) [inline src/schedule.py:1]: @arjnpandey Review this file

File: src/sim_util.py [hunk 1]
```diff
@@ -5,8 +5,11 @@
 import matplotlib.pyplot as plt
```
Comment (fazerlicourice7) [inline src/sim_util.py:5]: @daheehee review this file

File: src/symbolic_simulate.py [hunk 2]
```diff
@@ -21,9 +21,15 @@
 import sim_util
```
Comment (fazerlicourice7) [inline src/symbolic_simulate.py:21]: @dwoen Review this file

File: src/simulate.py [hunk 3]
```diff
@@ -1,12 +1,9 @@
 # builtin modules
```
Comment (fazerlicourice7) [inline src/simulate.py:1]: @dannyslowpark review this file

File: src/codesign.py [hunk 4]
```diff
@@ -2,6 +2,7 @@
 import os
```
Comment (fazerlicourice7) [inline src/codesign.py:2]: @console-beaver review this file

**Comment by fazerlicourice7** [comment]
@dwoen @daheehee @dannyslowpark @arjnpandey @console-beaver 

Look at the github docs here for general PR info: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews

Here for what to look for in acode review: https://google.github.io/eng-practices/review/reviewer/looking-for.html

I want all of you to at least review the one file I mentioned you in, feel free to review other files as well.

**Comment by fazerlicourice7** [comment]
Something to look out for in particular is random pieces of commented code. We need to clean the repo of random nonsense that doesn't get used.

File: src/arch_search_util.py
Comment (arjnpandey) [inline src/arch_search_util.py:1]: style wise everything looks good. could consider adding some in-line comments to better understand logic.

**Review by arjnpandey — APPROVED** [review]
looks good

**Pull request merged**